src/components/Attachment/Attachment.js

```diff
@@ -1,264 +1,244 @@
-    /**
-     * Custom UI component for attachment icon for type 'file' attachment.
-     * Defaults to: https://github.com/GetStream/stream-chat-react-native/blob/master/src/components/FileIcon.js
-     */
-    AttachmentFileIcon: PropTypes.oneOfType([
-      PropTypes.node,
-      PropTypes.elementType,
-    ]),
-    /**
-     * Custom UI component to display image attachments.
-     * Defaults to https://github.com/GetStream/stream-chat-react-native/blob/master/src/components/Gallery.js
-     */
-    Gallery: PropTypes.oneOfType([PropTypes.node, PropTypes.elementType]),
-    /**
-     * Custom UI component to display generic media type e.g. giphy, url preview etc
-     * Defaults to https://github.com/GetStream/stream-chat-react-native/blob/master/src/components/Card.js
-     */
-    Card: PropTypes.oneOfType([PropTypes.node, PropTypes.elementType]),
-    /**
-     * Custom UI component to override default header of Card component.
-     * Accepts the same props as Card component.
-     */
-    CardHeader: PropTypes.oneOfType([PropTypes.node, PropTypes.elementType]),
-    /**
-     * Custom UI component to override default cover (between Header and Footer) of Card component.
-     * Accepts the same props as Card component.
-     */
-    CardCover: PropTypes.oneOfType([PropTypes.node, PropTypes.elementType]),
-    /**
-     * Custom UI component to override default Footer of Card component.
-     * Accepts the same props as Card component.
-     */
-    CardFooter: PropTypes.oneOfType([PropTypes.node, PropTypes.elementType]),
-    /**
-     * Custom UI component to display attachment actions. e.g., send, shuffle, cancel in case of giphy
-     * Defaults to https://github.com/GetStream/stream-chat-react-native/blob/master/src/components/AttachmentActions.js
-     */
-    AttachmentActions: PropTypes.oneOfType([
-      PropTypes.node,
-      PropTypes.elementType,
-    ]),
-  };
-  static defaultProps = {
-    AttachmentFileIcon: FileIcon,
+const Attachment = (props) => {
+  const {
+    actionHandler,
+    alignment,
+    attachment,
     AttachmentActions,
-    Gallery,
     Card,
+    CardCover,
+    CardFooter,
+    CardHeader,
     FileAttachment,
-  };
+    Gallery,
+  } = props;
 
-  constructor(props) {
-    super(props);
+  if (!attachment) {
+    return null;
   }
 
-  render() {
-    const {
-      attachment: a,
-      Gallery,
-      Card,
-      CardHeader,
-      CardCover,
-      CardFooter,
-      FileAttachment,
-      AttachmentActions,
-    } = this.props;
-    if (!a) {
-      return null;
-    }
-
-    const Giphy = this.props.Giphy ? this.props.Giphy : Card;
-    const UrlPreview = this.props.UrlPreview ? this.props.UrlPreview : Card;
+  const Giphy = props.Giphy ? props.Giphy : Card;
+  const UrlPreview = props.UrlPreview ? props.UrlPreview : Card;
+  const cardProps = {
+    Header: CardHeader ? CardHeader : undefined,
+    Cover: CardCover ? CardCover : undefined,
+    Footer: CardFooter ? CardFooter : undefined,
```

Comment:
would these all not be undefined anyways if CardHeader or whatever doesn't exist?